src/xtgeo/grid3d/_grid_etc1.py

```diff
@@ -290,6 +291,91 @@ def get_heights_above_ffl(
     return htop, hbot, hmid
 
 
+def get_property_between_surfaces(
+    grid: Grid,
+    top: RegularSurface,
+    base: RegularSurface,
+    value: int | float = 1,
+    name: str = "between_surfaces",
+) -> GridProperty:
+    """For a grid, create a grid property with value <value> between two surfaces.
+
+    The value would be zero elsewhere, or if surfaces has inactive nodes.
+    """
+
+    grid._xtgformat2()
+    logger.debug("Creating property between surfaces...")
+
+    xmid, ymid, zmid = _internal.grid3d.grid_cell_centers(
+        grid.ncol,
+        grid.nrow,
+        grid.nlay,
+        grid._coordsv,
+        grid._zcornsv,
+        grid._actnumsv,
+        True,  # asmasked, will give Nan for inactive cells; intentional here
+    )
+
+    top_ = top
+    base_ = base
+    if top.yflip == -1:
+        top_ = top.copy()
+        top_.make_lefthanded()
+        logger.debug("Top surface is right-handed, flipping a copy prior to operation")
+    if base.yflip == -1:
+        base_ = base.copy()
+        base_.make_lefthanded()
+        logger.debug("Base surface is right-handed, flipping a copy prior to operation")
+
+    diff = base_ - top_
+    if diff.values.mean() <= 0:
+        raise ValueError("Top surface must be above base surface")
+
+    # array is always 0, 1 integer
+    array = _internal.grid3d.grid_assign_value_between_surfaces(
+        grid.ncol,
+        grid.nrow,
+        grid.nlay,
+        xmid,
+        ymid,
+        zmid,
+        top_.ncol,
+        top_.nrow,
+        top_.xori,
+        top_.yori,
+        top_.xinc,
+        top_.yinc,
+        top_.rotation,
+        top_.values.filled(np.nan),
+        base_.ncol,
+        base_.nrow,
+        base_.xori,
+        base_.yori,
+        base_.xinc,
+        base_.yinc,
+        base_.rotation,
+        base_.values.filled(np.nan),
+    )
+
+    is_discrete = True
+
+    if isinstance(value, float):
+        array = array.astype(float) * value
```

Comment:
If array is an `py::array<bool>` I _think_ this should be a bit more performant as well